### PR TITLE
CETS backend for mongoose_cluster_id (minimal logic)

### PR DESCRIPTION
**Pull request by arcusfelis**
This PR addresses "MIM-2040 System metrics fail with no_client_id".

This is a copy of https://github.com/esl/MongooseIM/pull/4135 but the difference is that this branch does not try to fix all the tests, that still assume Mnesia is started.

Proposed changes include:
* CETS backend for mongoose_cluster_id.
* `Added cets:insert_serial/2`. See PR for CETS library: https://github.com/esl/cets/pull/33
* _Actually ensure that Mnesia is stopped after applying preset. This requires some fixes in distributed_helper, because some functions assume Mnesia is always running._  - update: we have the code but we disable this ensuring check for now, because it breaks a lot of tests. We will try to fix tests suite-by-suite and enable this check..

### CODE REVIEW

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_25](https://circleci.com/gh/esl/MongooseIM/193576) / elasticsearch_and_cassandra_mnesia / 359703b7511eefdd46a9393924d35f3f0004eb8c
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4136/193576/elasticsearch_and_cassandra_mnesia.25.3-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4136/193576/elasticsearch_and_cassandra_mnesia.25.3-amd64/big/ct_run.test%40da524e068ede.2023-10-04_10.41.09/index.html)
**OK: 370** / Failed: 0 / **User-skipped: 38** / Auto-skipped: 0

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_25](https://circleci.com/gh/esl/MongooseIM/193637) / elasticsearch_and_cassandra_mnesia / 16e8d99dc71ff5b43ed88644972779f24fdeb053
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4136/193637/elasticsearch_and_cassandra_mnesia.25.3-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4136/193637/elasticsearch_and_cassandra_mnesia.25.3-amd64/big/ct_run.test%4092b4518be37d.2023-10-04_10.59.47/index.html)
**OK: 370** / Failed: 0 / **User-skipped: 38** / Auto-skipped: 0

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_25](https://circleci.com/gh/esl/MongooseIM/193700) / elasticsearch_and_cassandra_mnesia / 7e3de03d767a2a8724f15a7d8e92d2fee05512f3
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4136/193700/elasticsearch_and_cassandra_mnesia.25.3-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4136/193700/elasticsearch_and_cassandra_mnesia.25.3-amd64/big/ct_run.test%40dce059b2112d.2023-10-04_13.01.36/index.html)
**OK: 370** / Failed: 0 / **User-skipped: 38** / Auto-skipped: 0

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_25](https://circleci.com/gh/esl/MongooseIM/193746) / elasticsearch_and_cassandra_mnesia / 3affe01f0b4ae28d8a4f791befd140c1454d03cc
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4136/193746/elasticsearch_and_cassandra_mnesia.25.3-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4136/193746/elasticsearch_and_cassandra_mnesia.25.3-amd64/big/ct_run.test%4013bf1a8d90dd.2023-10-04_13.27.36/index.html)
**OK: 369** / Failed: 0 / **User-skipped: 38** / Auto-skipped: 0

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_25](https://circleci.com/gh/esl/MongooseIM/193768) / elasticsearch_and_cassandra_mnesia / 106d32905111386b3b6fc924fc9338cb47e0f3b1
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4136/193768/elasticsearch_and_cassandra_mnesia.25.3-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4136/193768/elasticsearch_and_cassandra_mnesia.25.3-amd64/big/ct_run.test%40e9bc5d6e8dc9.2023-10-04_13.42.10/index.html)
**OK: 369** / Failed: 0 / **User-skipped: 38** / Auto-skipped: 0

**Review by chrzaszcz — COMMENTED** [review]
Looks ok in general, I added some comments.

File: src/mongoose_internal_databases.erl [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+            []
+    end ++
+    case mongoose_config:lookup_opt([internal_databases, cets]) of
+        {ok, _} ->
+            [cets];
+        {error, _} ->
+            []
+    end.
+
+running_internal_databases() ->
```
[the remark this answers is not in the document]
Comment (arcusfelis) [reply src/mongoose_internal_databases.erl:50]: I needed `actually_running_internal_databases` function to compare both outputs as a sanity check before starting any big tests.

In the perfect world running_internal_databases is not only a subset of configured_internal_databases, but have to match (maybe mnesia fails to start case would ruin the perfect world intentions).

File: big_tests/tests/mongoose_sanity_checks_SUITE.erl [hunk 2]
```diff
@@ -1,10 +1,18 @@
 -module(mongoose_sanity_checks_SUITE).
 
 -compile([export_all, nowarn_export_all]).
+-import(distributed_helper, [mim/0, rpc/4]).
+-include_lib("eunit/include/eunit.hrl").
```
[the remark this answers is not in the document]
Comment (arcusfelis) [reply big_tests/tests/mongoose_sanity_checks_SUITE.erl:5]: oh, nice catch, there was a sanity check for internal databases. I could revert changed to this module :)

**Comment by mongoose-im** [comment]
[small_tests_25_arm64](https://circleci.com/gh/esl/MongooseIM/194295) / small_tests / 48f1203989fc98de3266529b4e57495b17d22aad
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4136/194295/small_tests.25.3-arm64) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4136/194295/small_tests.25.3-arm64/small/ct_run.mongooseim@localhost.2023-10-06_11.09.27/lib.mongooseim.logs/run.2023-10-06_11.09.28/suite.log.html)

**Review by chrzaszcz — APPROVED** [review]
👌

**Pull request merged**